File: SnapHack/context/samples/Spectacles Mobile Kit/sdk/iOS/SpectaclesKit/Sources/Implementation/Bonding/SpectaclesBonding.swift
// Copyright © 2024 Snap, Inc. All rights reserved.

import Foundation

struct BondingIdentifier: Equatable {
    /// The bonding ID assigned by the Spectacles when creating a bonding
    let assignedId: UUID
    /// BLE address of the Spectacles
    let deviceAddress: UUID

    enum Keys: String {
        case assignedId = "bondingId"
        case deviceAddress = "address"
    }
}

struct SpectaclesBonding: Bonding, Equatable {
    public let id: String

    let identifier: BondingIdentifier
    let deviceId: String
    let lensId: String
    var publicKey: SecKey?
    var keyAlgorithm: String?
}

File: SnapHack/context/samples/Spectacles Mobile Kit/sdk/Android/SpectaclesKit/kit/src/main/java/com/snap/spectacles/kit/core/SpectaclesBonding.kt
package com.snap.spectacles.kit.core

import com.snap.spectacles.kit.SpectaclesKit

/**
 * A hybrid identifier that can be used to identify a Spectacles bonding.
 * The [assignedId] is a unique identifier assigned by the Spectacles when creating a bonding, but
 * it may not be unique across different Spectacles.
 * The [deviceAddress] is the BLE address of the Spectacles, which is unique.
 */
data class BondingIdentifier(
    // The bonding id assigned by the Spectacles when creating a bonding
    val assignedId: String,
    // BLE address of the Spectacles
    val deviceAddress: String,
    val deviceId: String,
)

/**
 * Represents a Spectacles bonding.
 */
data class SpectaclesBonding(
    override val id: String,
    val identifier: BondingIdentifier,
    val requested: SpectaclesKit.BondingRequest
) : SpectaclesKit.Bonding

typealias SpectaclesBondingSerializer = Serializer<SpectaclesBonding, String>
